src/test/java/tests/RepositorySelectorPopulationTests.java

```diff
@@ -47,16 +49,17 @@ public void noJsonFiles() {
     public void oneInvalidJsonFile() throws IOException {
         File invalidJson = new File("store/test/dummy-dummy.json");
         assert invalidJson.createNewFile();
-        RepoIO testIO = new RepoIO(true, true);
+        RepoIO testIO = TestController.createTestingRepoIO(Optional.of(new JSONStoreStub()));
```

Comment:
Wouldn't we want to use the normal `JSONStore` here? Since we are testing the JSON operations.

I am just ensuring the consistency since all the tests in this file used `RepoIO(true, true)` but you changed some to use `JSONStoreStub()` and some to `Optional.empty()`. Is there a reason behind that?